lib/host.js

```diff
@@ -305,7 +335,47 @@ HostConnectionPool.prototype.forceShutdown = function () {
   });
 };
 
+/**
+ * Closes all open connections and schedules reconnection
+ * @param {Number} delay
+ */
+HostConnectionPool.prototype.scheduleReconnection = function (delay) {
+  this._forceShutdown();
+  this.reconnectionTimeout = setTimeout(this._attemptReconnection.bind(this), delay);
+};
+
+/**
+ * Prevents reconnection timeout from triggering
+ */
+HostConnectionPool.prototype.clearReconnection = function () {
+  if (!this.reconnectionTimeout) {
+    return;
+  }
+  clearTimeout(this.reconnectionTimeout);
+  this.reconnectionTimeout = null;
+};
+
+HostConnectionPool.prototype._attemptReconnection = function () {
+  this.reconnectionTimeout = null;
+  var c = this._createConnection();
+  var self = this;
+  c.open(function (err) {
+    if (err) {
+      self.log('info', util.format('Reconnection attempt to host %s failed', self.address), err);
+      self.host.setDown();
+      return c.close();
+    }
+    self.host.setUp();
```

Comment:
I think what @gwicke described was related to using the user call to make reconnection attempts (in their case to dead nodes).

As you suggest, after a succeeded reconnection attempt we can call `_maybeCreatePool()` to start growing additional core connections (if any).

About using the first connection if its already "growing", it also makes sense, the C# driver does the same thing.